src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/UWP.cs

```diff
@@ -38,24 +38,17 @@ public class UWP
 
         public UWP(Package package)
         {
-            Location = package.InstalledLocation.Path;
+            
             Name = package.Id.Name;
             FullName = package.Id.FullName;
             FamilyName = package.Id.FamilyName;
-            InitializeAppInfo();
-            Apps = Apps.Where(a =>
-            {
-                var valid =
-                    !string.IsNullOrEmpty(a.UserModelId) &&
-                    !string.IsNullOrEmpty(a.DisplayName);
```

Comment:
Is there a reason for removing this condition? Has it been moved somewhere else? This might cause invalid apps to be displayed.

Reply:
> Is there a reason for removing this condition? Has it been moved somewhere else? This might cause invalid apps to be displayed.

Thats a good catch. I meant to move it to the initialize app info.    I need to abstract the app initialization somehow to be able to properly unit test the packagerepository.  That was the original intent of moving this outside the constructor, but it still isn't testable as is.

Reply:
Also.  `Package.InstalledLocation.Path` throws a COM exception when given the package from the OnUninstall callback.   I wanted to be able to construct a UWP object for comparison sake without having to initialize it.